lib/chef/policy_builder/policyfile.rb

```diff
@@ -356,6 +354,28 @@ def config
         Chef::Config
       end
 
+      private
+
+      def compat_mode_manifest_for(cookbook_name, lock_data)
+        xyz_version = lock_data["dotted_decimal_identifier"]
+        http_api.get("cookbooks/#{cookbook_name}/#{xyz_version}")
+      rescue Exception => e
+        message = "Error loading cookbook #{cookbook_name} at version #{xyz_version}: #{e.class} - #{e.message}"
```

Comment:
It would help to differentiate the error messages between these 2 methods - I say add a `...at version #{xyz_version} from compatibility api:...`